src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -105,7 +108,7 @@ public async Task WatchAsync(DotNetWatchContext context, CancellationToken cance
                     while (true)
                     {
                         fileSetTask = fileSetWatcher.GetChangedFileAsync(combinedCancellationSource.Token);
-                        finishedTask = await Task.WhenAny(processTask, fileSetTask, cancelledTaskSource.Task);
+                        finishedTask = await Task.WhenAny(processTask, fileSetTask).WaitAsync(combinedCancellationSource.Token);
```

Comment:
Using cool new API. It works out nicely because it throws when cancellation is triggered and exits multiple while loops.